server/controllers/Article.js

```diff
@@ -0,0 +1,217 @@
+              });
+            })
+            .catch(next);
+        } else {
+          res.status(404).json({
+            status: 'fail',
+            message: 'Article not found',
+          });
+        }
+      })
+      .catch(next);
+  };
+  
+  static updateArticle(req, res, next) {
+    const {
+      title, description, body,
+    } = req.body.article;
+  
+    Article.find({ where: { slug: req.params.slug } })
+      .then((article) => {
+        if (article) {
+          const updatedArticle = article;
+          updatedArticle.title = title !== undefined ? title : article.title;
+          updatedArticle.description = description !== undefined ? description : article.description;
+          updatedArticle.body = body !== undefined ? body : article.body;
+          updatedArticle.save();
+          User.find({ where: { id: article.authorId } })
+            .then((user) => {
+              res.status(200).json({
+                article: {
+                  slug: article.slug,
+                  title: article.title,
+                  description: article.description,
+                  body: article.body,
+                  author: user.username,
+                },
+                message: 'Article Updated',
+              });
+            })
+            .catch(next);
+        } else {
+          res.status(404).json({
+            status: 'fail',
+            message: 'Article not found',
+          });
+        }
+      })
+      .catch(next);
+  };
+  
+  static deleteArticle(req, res, next) {
+    Article.find({ where: { slug: req.params.slug } })
+      .then((article) => {
+        if (article) {
+          Article.destroy({ where: { slug: req.params.slug } })
+            .then((success) => {
+              if (success) {
+                res.status(200).json({
+                  article: {
+                    slug: article.slug,
+                    title: article.title,
+                    description: article.description,
+                    body: article.body,
+                  },
+                  message: 'Article Deleted',
+                });
+              } else {
+                res.status(500).json({
+                  status: 'fail',
+                  message: 'Could not delete Article',
+                });
+              }
+            })
+            .catch(next);
+        } else {
+          res.status(404).json({
+            status: 'fail',
+            message: 'Article not found',
+          });
+        }
+      })
+      .catch(next);
+  };
+  
+  static rateArticle(req, res) {
+    let { rating } = req.query;
+    let { articleId } = req.params;
+    let userId = req.body.decoded.id;
+    rating = parseInt(rating, 10);
+    articleId = parseInt(articleId, 10);
+    userId = parseInt(userId, 10);
+    return Article
+      .findById(articleId)
+      .then((article) => {
+        if (article.dataValues) {
+          if (article.dataValues.authorId === parseInt(userId, 10)) {
```

Comment:
I believe you can call `article.authorId` directly, I might be wrong. But try it out.